src/config.rs

```diff
@@ -87,6 +87,8 @@ pub struct Config {
 /// Helper struct that handles some specific field deserialization for `Config` struct
 struct ConfigDeserializer;
 
+type CriticalityString = (Criticality, String);
```

Comment:
The tuple seems simple enough, right? Why create a new alias? it could make it more difficult to understand what are we working with.

Reply:
Clippy told me that `result::Result<(Criticality, String), D::Error>` would be too long, so I split that up.